### PR TITLE
[Choreo] Fix intelligent routing availability after deleting and re-creating an API

### PR DESCRIPTION
**Pull request by chathuranga95**
### Purpose
This PR,
- Removes unwanted minor version based intelligent routing regex building for the envoy router
- Fixes a bug where the vhost is neglected on API undeployment

### Issues
Addresses: https://github.com/wso2-enterprise/choreo/issues/28419

### Automation tests
 - Unit tests added: No
 - Integration tests added: No

### Tested environments

OS: ubuntu
Go: 1.22

---
#### Maintainers: Check before merge
- [ ] Assigned 'Type' label
- [ ] Assigned the project
- [ ] Validated respective github issues
- [ ] Assigned milestone to the github issue(s)

### CODE REVIEW

**Review by VirajSalaka — APPROVED** [review]

File: adapter/internal/discovery/xds/semantic_versioning.go [hunk 1]
```diff
@@ -36,34 +36,14 @@ func GetVersionMatchRegex(version string) string {
 func GetMajorMinorVersionRangeRegex(semVersion semantic_version.SemVersion) string {
 	majorVersion := strconv.Itoa(semVersion.Major)
 	minorVersion := strconv.Itoa(semVersion.Minor)
-	if semVersion.Patch == nil {
```
Comment (VirajSalaka) [inline adapter/internal/discovery/xds/semantic_versioning.go:39]: shall we remove Patch property from the struct as well.

Comment (chathuranga95) [reply adapter/internal/discovery/xds/semantic_versioning.go:39]: Done with: https://github.com/wso2/product-microgateway/pull/3506/commits/f7728c1c1ba607eb752bc325a389bb645f7b37b1

**Review by VirajSalaka — APPROVED** [review]

**Comment by choreo-cicd** [comment]
[succeeded]  : dev-deployment-v2 : 20240510.15

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : dev-deployment-v2 : 20240510.15

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(EastUS) cluster : dev-deployment-v2 : 20240510.15

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(NorthEU) cluster : dev-deployment-v2 : 20240510.15

**Comment by choreo-cicd** [comment]
[failed] Dataplane(EastUS) cluster : stage-deployment-v2 : 20240513.3

**Comment by choreo-cicd** [comment]
[failed] Controlplane cluster : stage-deployment-v2 : 20240513.3

**Comment by choreo-cicd** [comment]
[failed] Dataplane(NorthEU) cluster : stage-deployment-v2 : 20240513.4

**Comment by choreo-cicd** [comment]
[failed] Dataplane(CDP2-WU2) cluster : stage-deployment-v2 : 20240513.4

**Comment by choreo-cicd** [comment]
[failed] Dataplane(EastUS) cluster : stage-deployment-v2 : 20240513.4

**Comment by choreo-cicd** [comment]
[failed] Dataplane(CDP2-WU2) cluster : stage-deployment-v2 : 20240513.5

**Comment by choreo-cicd** [comment]
[failed] Dataplane(EastUS) cluster : stage-deployment-v2 : 20240513.5

**Comment by choreo-cicd** [comment]
[] Dataplane(NorthEU) cluster : stage-deployment-v2 : 20240513.6

**Comment by choreo-cicd** [comment]
[] Controlplane cluster : stage-deployment-v2 : 20240513.6

**Comment by choreo-cicd** [comment]
[] Dataplane(CDP2-WU2) cluster : prod-deployment-v2 : 20240515.2

**Comment by choreo-cicd** [comment]
[] Dataplane(CDP2-WU2) cluster : prod-deployment-v2 : 20240515.2

**Comment by choreo-cicd** [comment]
[] Dataplane(EastUS) cluster : prod-deployment-v2 : 20240515.2

**Comment by choreo-cicd** [comment]
[] Controlplane cluster : prod-deployment-v2 : 20240515.2

**Comment by choreo-cicd** [comment]
[] Controlplane cluster : prod-deployment-v2 : 20240515.2

**Comment by choreo-cicd** [comment]
[] Dataplane(NorthEU) cluster : prod-deployment-v2 : 20240515.2

**Comment by choreo-cicd** [comment]
[] Dataplane(NorthEU) cluster : prod-deployment-v2 : 20240515.2

**Pull request merged**